runtime/compiler/env/VMJ9.cpp

```diff
@@ -4348,12 +4348,24 @@ TR_J9VMBase::canDereferenceAtCompileTime(TR::SymbolReference *fieldRef, TR::Comp
             // ourselves to fold instance fields only in classes classes where
             // this is known to be safe.
 
-            J9Class *fieldClass = (J9Class*)fieldRef->getOwningMethod(comp)->getClassFromFieldOrStatic(comp, fieldRef->getCPIndex());
-            if (!fieldClass)
-               return false;
-
+            const char* name;
             int32_t len;
-            char * name = getClassNameChars((TR_OpaqueClassBlock*)fieldClass, len);
+
+            // Get class name for fabricated java field
+            if (fieldRef->getCPIndex() < 0 &&
+               fieldRef->getSymbol()->getRecognizedField() != TR::Symbol::UnknownField)
```

Comment:
Is it better to just use the `isFabricatedJavaField` created below?

Reply:
This function was used for VarHandle, with this change, it is not needed any more as the checks are pretty simple. I removed this function and replace it with similar code as above in `J9TransformUtil`

Reply:
This function was used for VarHandle, with this change, it is not needed any more as the checks are pretty simple. I removed this function and replace it with similar code as above in `J9TransformUtil`